app/views/play/modal/PlayItemsModal.coffee

```diff
@@ -212,6 +214,45 @@ module.exports = class PlayItemsModal extends ModalView
       @$el.one 'click', (e) ->
         button.removeClass('confirm').text($.i18n.t('play.unlock')) if e.target isnt button[0]
 
+  onSellButtonClicked: (e) ->
+    e.stopPropagation()
+    button = $(e.target).closest('button')
+    item = @idToItem[button.data('item-id')]
+    
+    # Prevent further action if the item cannot be sold
+    if !item.sellable()
+      return
+    
+    if button.hasClass('confirm')
+      # Sell logic here
+      sale = Sale.makeFor(item)
+      sale.save()
+    
+      #- set local changes to mimic what should happen on the server...
+      sales = me.get('sales') ? {}
+      sales.items ?= []
+      sales.items.push(item.get('original'))
+    
+      purchased = me.get('purchased') ? {}
+      purchased.items ?= []
+      _.pull(purchased.items, item.get('original'))
+      me.set('purchased', purchased)
+    
+      item.owned = false
+      me.set('sales', sales)
+      me.set('sold', (me.get('sold') ? 0) + item.sellPrice())
+
+      #- ...then rerender key bits
+      @renderSelectors(".item[data-item-id='#{item.id}']", "#gems-count")
+      @itemDetailsView.render()
+
```

Comment:
The item images are stored in the db object and need to be applied to the re-rendered icon. The logic is in onScrollItemPane and should probably be refactored out into a separate function so it can be used here and there.